cypress/e2e/view-filtering-datetime.cy.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+let localUser
+const tableTitle = 'View datetime filtering test table'
+
+const today = new Date()
+const [tomorrow, yesterday, daysAhead30, daysAhead60, daysAgo30, daysAgo60] = [1, -1, 30, 60, -30, -60].map(days => {
+	const d = new Date()
+	d.setDate(d.getDate() + days)
+	return d
+})
+
+const formatDate = (date) => date.toISOString().split('T')[0]
```

Comment:
I imagine that this test breaks if we use a different timezone? Should be fine, since it's just a test. Just confirming that the server logic itself is timezone-proof